prob question for QBasedPolicy

I find that when computing [prob(p::QBasedPolicy, env::AbstractEnv, action)](https://github.com/JuliaReinforcementLearning/ReinforcementLearning.jl/blob/c7b66e47e939592045326485b9c89b1486df165d/src/ReinforcementLearningCore/src/policies/q_based_policies/q_based_policy.jl#L37), it will use [prob(s::GreedyExplorer, values)](https://github.com/JuliaReinforcementLearning/ReinforcementLearning.jl/blob/c7b66e47e939592045326485b9c89b1486df165d/src/ReinforcementLearningCore/src/policies/q_based_policies/explorers/epsilon_greedy_explorer.jl#L205), which use `Categorical` to wrap the `prob` in the end. So that it would raise `AssertionError: length(A) == length(P)`, just like the following example:

```julia
using Flux
using ReinforcementLearning

# Wrap the AbstractEnv for inputting the NeuralNetworkApproximator
env = KuhnPokerEnv()
states = [
    (), (:J,), (:Q,), (:K,),
    (:J, :bet), (:J, :pass), (:Q, :bet), (:Q, :pass), (:K, :bet), (:K, :pass),
    (:J, :pass, :bet), (:J, :bet, :bet), (:J, :bet, :pass), (:J, :pass, :pass),
    (:Q, :pass, :bet), (:Q, :bet, :bet), (:Q, :bet, :pass), (:Q, :pass, :pass),
    (:K, :pass, :bet), (:K, :bet, :bet), (:K, :bet, :pass), (:K, :pass, :pass),
    (:J, :pass, :bet, :pass), (:J, :pass, :bet, :bet), (:Q, :pass, :bet, :pass),
    (:Q, :pass, :bet, :bet), (:K, :pass, :bet, :pass), (:K, :pass, :bet, :bet),
] # all states for players 1 & 2

states_indexes_Dict = Dict((i, j) for (j, i) in enumerate(states))

wrapped_env = StateTransformedEnv(
        env;
        state_mapping = s -> 
            current_player(env) == chance_player(env) ? s : [states_indexes_Dict[s]]
    )

# ignore the chance_player
while current_player(env) == chance_player(env)
    env |> action_space |> rand |> env
end

policy = QBasedPolicy(
    learner = BasicDQNLearner(
        approximator = NeuralNetworkApproximator(
            model = Dense(1, 2),
            optimizer = Descent(0.01),
        )
    ),
    explorer = GreedyExplorer(),
)

prob(policy, wrapped_env, 1)
```

In this example, 

```julia
julia> A = action_space(wrapped_env)
Base.OneTo(2)

julia> P = prob(policy, wrapped_env)
Distributions.Categorical{Float64, Vector{Float64}}(support=Base.OneTo(2), p=[1.0, 0.0])

julia> length(A), length(P)
(2, 1)
```

--- a/docs/homepage/blog/ospp_report_210370190/index.md
+++ b/docs/homepage/blog/ospp_report_210370190/index.md
@@ -401,7 +401,7 @@ Plus on the [`stop_condition`](https://github.com/JuliaReinforcementLearning/Rei
 
 \dfig{body;JuliaRL_MADDPG_KuhnPoker.png;Result of the experiment.}
 
-However, `KuhnPoker` is not good choice to show the performance of **MADDPG**. For testing the performance of the algorithm, I add [`SpeakerListenerEnv`](https://juliareinforcementlearning.org/docs/rlenvs/#ReinforcementLearningEnvironments.SpeakerListenerEnv-Tuple{}) into [ReinforcementLearningEnvironments.jl](https://juliareinforcementlearning.org/docs/rlenvs), which is a simple cooperative multi-agent game.
+However, `KuhnPoker` is not a good choice to show the performance of **MADDPG**. For testing the algorithm, I add [`SpeakerListenerEnv`](https://juliareinforcementlearning.org/docs/rlenvs/#ReinforcementLearningEnvironments.SpeakerListenerEnv-Tuple{}) into [ReinforcementLearningEnvironments.jl](https://juliareinforcementlearning.org/docs/rlenvs), which is a simple cooperative multi-agent game.
 
 Since two player have different dimensions of state and action in the `SpeakerListenerEnv`, the policy and the trajectory are customized as below:
 ```Julia
@@ -472,4 +472,4 @@ agents = MADDPGManager(
 
 Add the [`stop_condition`](https://github.com/JuliaReinforcementLearning/ReinforcementLearning.jl/blob/master/docs/experiments/experiments/Policy%20Gradient/JuliaRL_MADDPG_SpeakerListener.jl#L120) and designed [`hook`](https://github.com/JuliaReinforcementLearning/ReinforcementLearning.jl/blob/master/docs/experiments/experiments/Policy%20Gradient/JuliaRL_MADDPG_SpeakerListener.jl#L15), we can simply `run(agents, env, stop_condition, hook)` to run the experiment and use `Plots.plot` to the following result.
 
-\dfig{body;JuliaRL_MADDPG_SpeakerListener.png;Result of the experiment.}
+\dfig{body;JuliaRL_MADDPG_SpeakerListenerEnv.png;Result of the experiment.}

--- a/src/ReinforcementLearningCore/src/policies/q_based_policies/q_based_policy.jl
+++ b/src/ReinforcementLearningCore/src/policies/q_based_policies/q_based_policy.jl
@@ -2,6 +2,7 @@ export QBasedPolicy, TabularRandomPolicy
 
 using MacroTools: @forward
 using Flux
+using Distributions: Distribution, probs
 using Setfield: @set
 
 """
@@ -37,6 +38,9 @@ RLBase.prob(p::QBasedPolicy, env::AbstractEnv, ::FullActionSet) =
 function RLBase.prob(p::QBasedPolicy, env::AbstractEnv, action)
     A = action_space(env)
     P = prob(p, env)
+    if P isa Distribution
+        P = probs(P)
+    end
     @assert length(A) == length(P)
     if A isa Base.OneTo
         P[action]

